hare/eligibility/oracle.go

```diff
@@ -76,20 +83,28 @@ func (o *Oracle) BuildVRFMessage(id types.NodeId, layer types.LayerID, round int
 	return w.Bytes(), nil
 }
 
-// IsEligible checks if id is eligible on the given layer where msg is the VRF message, sig is the role proof and assuming commSize as the expected committee size
-func (o *Oracle) IsEligible(commSize uint32, id types.NodeId, layer types.LayerID, msg, sig []byte) (bool, error) {
+func (o *Oracle) activeSetSize(layer types.LayerID) uint32 {
+	fmt.Println("the epoc is: ", safeLayer(layer).GetEpoch(o.layersPerEpoch))
+	return o.activeSetProvider.ActiveSetSize(safeLayer(layer).GetEpoch(o.layersPerEpoch) - 1)
+}
+
+// Eligible checks if id is eligible on the given layer where msg is the VRF message, sig is the role proof and assuming commSize as the expected committee size
+func (o *Oracle) Eligible(layer types.LayerID, round int32, committeeSize int, id types.NodeId, sig []byte) (bool, error) {
```

Comment:
The fixed size oracle uses the pub
The real oracle is using the VRF
We can probably use only the VRF if we change the FSO.
I, however, prefer to keep it as is. I don't want to change the interface unless we are sure we don't need it.
If it is of concern to you we can open a technical debt to revisit this Q. Let me know if it is.

Reply:
It is indeed a concern. If the FSO can work only with the VRF key then we should definitely change the API. It is error-prone as someone can mistakenly use the pub for production code

Reply:
@barakshani please verify that I am not missing anything and that we only need the VRF key for the (real) oracle.
I remember that the message passed to the beacon uses both the ED and the VRF keys. But as much as I remember you said it is not necessary.

Reply:
As long as there's a one-to-one correspondence between VRF/ED keys (no VRF id with more than one ED id and vice versa), then it does not matter: we can use ED, VRF or (ED,VRF).

The message should include the "identity" of the miner. We define id = (ED,VRF), so if we use the notion of id in other places (like ATX, NIPST etc.), I suggest to keep it the way it is, i.e. both keys, for consistency.

Reply:
Yes, I agree.

@zalmen - ok?

Reply:
On a second thought, it is not clear that we need the id at all, since it is linked to the signature anyway. For comparison, the VRF block oracle does not take 'id' as input.
We should clarify this point.

Reply:
@gavraz @barakshani  - I'm not arguing against having the tuple as the node id. My problem was that in some places we used the id structure after only setting one of its variants.